src/ResourceManager/MarketplaceOrdering/Commands.MarketplaceOrdering/Cmdlets/Agreements/GetAzureRmMarketplaceTerms.cs

```diff
@@ -0,0 +1,54 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.MarketplaceOrdering.Common;
+using Microsoft.Azure.Commands.MarketplaceOrdering.Models;
+using Microsoft.Azure.Management.MarketplaceOrdering;
+using Microsoft.Azure.Management.MarketplaceOrdering.Models;
+
+namespace Microsoft.Azure.Commands.MarketplaceOrdering.Cmdlets.Agreements
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmMarketplaceTerms", DefaultParameterSetName = Constants.ParameterSetNames.AgreementParameterSet), OutputType(typeof(PSAgreementTerms))]
```

Comment:
@prbansa I missed the design review for these cmdlets, so this may have been answered: is this a resource that can be created for a specific subscription, or even a resource group? Most `Get-*` cmdlets support providing no parameters (returns all resources in the subscription) or providing a resource group (returns all resources in the resource group).

Reply:
@cormacpayne - Get-AzureRmMarketplaceTerms returns "Terms" object which user can accept and fully automate 3rd party VM deployment experience.
This is not a resource so Get-* is not supported in this case.